bluebutton-data-model-rif/src/main/resources/db/scripts/Beneficiaries_with_duplicate_hicns.sql

```diff
@@ -0,0 +1,108 @@
+/*
+ * Queries for finding HICN and HICN hash collisions between beneficiary records.
+ *
+ * These queries are intended to be run manually, by copy-pasting them into an
+ * interactive `psql` session, or somesuch.
+ */
+
+-- How many "hicn"s are associated with more than one "beneficiaryId", whether or not those
+-- "beneficiaryId"s resolve to an actual "Beneficiaries" record?
+--
+-- On 2019-06-07, this was run in PROD:
+-- * returned 2,697,057 rows
+-- * with a max count of 4
+-- * in 1478941.350 ms (0:25h)
+WITH
+  hicn_id_tuples AS (
+    SELECT
+        DISTINCT "beneficiaryId" AS bene_id, "hicn" AS hicn
+      FROM "Beneficiaries"
+    UNION ALL
```

Comment:
Got a phone review from @dho937 that pointed out a major bug here: using `UNION ALL` leads to duplicates and other weirdness:

* Consider this data:
    
    ```
    1, 1A
    2, 2A
    UNION ALL
    1, 1A
    2, 2A
    3, 3A
    4, 1A
    ```
    
* Instead of getting an actual summed-`count(...)` of all the distinct-and-duplicate tuples, I'll get a count of _rows_ with each HICN, i.e. a count of 3 for the HICN "1A", instead of the correct count of 2.

This could be fixed either by switching to a `UNION` (which will de-dupe the CTE) or by switching all of the `count(bene_id)` expressions to `count(DISTINCT bene_id)`, instead.

Reply:
The fixed query returns 74,142 rows, with a max count of 4 (for just a single record), then a bunch of 3s and 2s.

```2019-06-25 10:12:46-04:00
fhirdb=# WITH
fhirdb-#   hicn_id_tuples AS (
fhirdb(#     SELECT
fhirdb(#         DISTINCT "beneficiaryId" AS bene_id, "hicn" AS hicn
fhirdb(#       FROM "Beneficiaries"
fhirdb(#     UNION
fhirdb(#     SELECT
fhirdb(#         DISTINCT "beneficiaryId" AS bene_id, "hicn" AS hicn
fhirdb(#       FROM "BeneficiariesHistory"
fhirdb(#   )
fhirdb-# SELECT
fhirdb-#     hicn, count(bene_id)
fhirdb-#   FROM hicn_id_tuples
fhirdb-#   GROUP BY hicn
fhirdb-#   HAVING count(bene_id) > 1
fhirdb-#   ORDER BY count(bene_id) DESC, hicn ASC;
Time: 2901512.386 ms```

Reply:
As I recall, that's the same result count that I got when I tried switching all the `count(bene_id)` expressions to `count(DISTINCT bene_id)`.